### PR TITLE
initial JSX proposal

### PR DESCRIPTION
**Pull request by clarkbw**

### CODE REVIEW

File: text/0005-jsx.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+### Existing Projects
+
+The [debugger.html](https://github.com/devtools-html/debugger.html) is currently considering using JSX.  The debugger has a [PR](https://github.com/devtools-html/debugger.html/pull/1712) that experiments with the change and an [issue](https://github.com/devtools-html/debugger.html/issues/1747) discussing the possibility.
+
+The [perf.html](https://github.com/devtools-html/perf.html) project had previously adopted JSX and plans to continue using it.
+
+### Working Group
+
+- @jasonLaster
+- ??
```
Comment (jasonLaster) [inline text/0005-jsx.md]: @gregtatum 
Maybe one other person who's in mc?

File: text/0005-jsx.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+JSX requires a build step, using something like webpack, so if we feel that a build step is not a requirement we are willing to take on then we should not adopt JSX and continue to use the `React.DOM` practices we currently have.  This build step require can continue to be a per-project decision.
+
+### Adoption
+
+With all new code and as we significantly refactor components of existing projects we should adopt JSX as the standard practice for our `render` methods when there is already a build step required for the project.
+
+### Metrics
+
+Significant metrics will be difficult to gather here. We should see more ownership of the components by less technical contributors and overall a simpler time getting people on boarded.
```
Comment (jasonLaster) [inline text/0005-jsx.md]: Maybe check in after 6 weeks and see if there are places where it doesn't make sense. 

Perhaps look at major components that are JSXd and compare them with components that have not been refactored.

File: text/0005-jsx.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+### Unresolved questions
+
+???
```
Comment (jasonLaster) [inline text/0005-jsx.md]: Where will the tools need to be changed? Linting?

File: text/0005-jsx.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+Adopt JSX as the default syntax for the `render` method of our react components.
+
+The current DevTools practice for implementing the `render` method of our react components uses the `React.DOM` interface.
+
+Where our current code might look like this:
+
+```js
+import React, { DOM as dom } from "react";
+class X extends React.Component {
```
Comment (gasolin) [inline text/0005-jsx.md]: class `CamelCase`

File: text/0005-jsx.md (hunk 3 above)
Comment (gasolin) [inline text/0005-jsx.md]: Maybe we could clarify which phase could adapt JSX, such as

When there's No build process, recommend to use Use `React.DOM`
* phase I deXUL
* phase II deChrome

When there's a build process, recommand to use Use `JSX`
* phase III host on Github

File: text/0005-jsx.md (hunk 1 above)
Comment (gregtatum) [inline text/0005-jsx.md]: I can be on it too. But definitely need someone else who works more in mc.

File: text/0005-jsx.md (hunk 3 above)
Comment (gregtatum) [inline text/0005-jsx.md]: What about shared components like the Tree component?

**Comment by jryans** [comment]
I noticed this [comment][1] from an observer on Twitter:

"@FirefoxDevTools are moving the tools to React/Redux. The part that makes me smile the most is that there is *NO* JSX. Someone else gets it."

Just thought I should record that here for reference.

[1]: https://twitter.com/toastal/status/837377895005831174

**Comment by jryans** [comment]
The changed files view is somewhat confusing here, since it includes various RFC setup commits that have already landed.

**Comment by juliandescottes** [comment]
@jasonLaster can you hit merge ? :)

**Pull request merged**